### PR TITLE
Add Knative API Specification

### PR DESCRIPTION
**Pull request by dgerd**
Adding a Markdown formatted version of the approved Knative API
Specification to the repository. Once this is checked in it will replace
the Google Doc found here:
https://docs.google.com/document/d/1UHT3W61tbX1qnEPA0qtoNUACki0d8FkXJ10JHOQu9uU/edit#

Fixes https://github.com/knative/serving/issues/4517
Related https://github.com/knative/serving/issues/4227

### CODE REVIEW

File: docs/serving/spec/knative-api-specification-1.0.md [hunk 1]
```diff
@@ -0,0 +1,2615 @@
...
+*   `metadata.labels` and `metadata.annotations` from the Configuration MUST NOT be copied to the newly-created Revision.
+
+Configuration MUST track the status of owned Revisions in order of creation, and report the name of the most recently created Revision in `status.latestCreatedRevisionName` and the name of the most recently created Revision where the `Ready` Condition is `True` in `status.latestReadyRevisionName`. These fields MAY be used by client software and Route objects to determine the status of the Configuration and the best target Revision for requests.
+
+
+## Revision
+
+A Revision is automatically scaled by the underlying infrastructure; a Revision with no referencing Routes MAY be automatically scaled to zero instances and backing resources collected. Additionally, Revisions which have been scaled to zero instances and which are older than the oldest live Revision (referenced by at least one Route or latest for the Configuration) MAY be automatically deleted by the system.
```
Comment (mattmoor) [inline docs/serving/spec/knative-api-specification-1.0.md]: nit: zero scale isn't a requirement for GC, only that things are orphaned.

File: docs/serving/spec/knative-api-specification-1.0.md [hunk 2]
```diff
@@ -0,0 +1,2615 @@
...
+   </td>
+   <td><strong>Schema Requirement</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>containers</code>
+   </td>
+   <td>[]<a href="#container">Container</a>
+(Required)
+Min: 1
```
Comment (mattmoor) [inline docs/serving/spec/knative-api-specification-1.0.md]: nit: `<br>` before (it wraps funny for me)

File: docs/serving/spec/knative-api-specification-1.0.md (hunk 1 above)
Comment (dgerd) [reply docs/serving/spec/knative-api-specification-1.0.md]: fixed

**Pull request merged**